File: data_structures/linked_lists/insert_tail_linked_list/sol.py
def Insert(head, data):
#    if not head:
#        head = Node(data)
#    elif not head.next:
#        head.next = Node(data)
#    else:
#       Insert(head.next, data)
#    return head

    if not head:
        return Node(data)
    return Node(head.data, Insert(head.next, data))

File: data_structures/linked_lists/insert_tail_linked_list/sol.java
/*
  Insert Node at the end of a linked list
  head pointer input could be NULL as well for empty list
  Node is defined as
  class Node {
     int data;
     Node next;
  }
*/
Node Insert(Node head,int data) {
// This is a "method-only" submission.
// You only need to complete this method.
    if(head == null) {
        head.data = data;
    } else {
    if(head.next == null) {
        head.next = new Node();
        head.next.data = data;
    } else{
        Insert(head.next, data);
    }
    }

    return head;

}

